src/Models/Laravel/Traits/EagerLoadSelectFields.php

```diff
@@ -296,18 +298,62 @@ public static function loadWallet(
 
                     break;
                 case 'tokenAccounts':
+                    $args = $args ?: Arr::get($selections, 'tokenAccounts.args');
+                    $filters = Arr::get($args, 'bulkFilter');
                     $withCount[$relation] = fn ($query) => $query->when(
-                        Arr::get($args, 'collectionIds'),
+                        empty($filters) && Arr::get($args, 'collectionIds'),
                         fn ($q) => $q->whereIn(
                             'collection_id',
                             DB::table('collections')->select('id')->whereIn('collection_chain_id', $args['collectionIds'])
                         )
                     )->when(
-                        Arr::get($args, 'tokenIds'),
+                        empty($filters) && Arr::get($args, 'tokenIds'),
                         fn ($q) => $q->whereIn(
                             'token_id',
                             DB::table('tokens')->select('id')->whereIn('token_chain_id', $args['tokenIds'])
                         )
+                    )->when(
+                        $filters,
+                        function ($query) use ($filters): void {
+                            $collections = Collection::whereIn('collection_chain_id', collect($filters)->pluck('collectionId'))
+                                ->pluck('id', 'collection_chain_id')
+                                ->all();
+
+                            $rangeRegx = '/-?[0-9]+(\.\.)-?[0-9]+/';
```

Comment:
We have a `HasIntegerRanges` trait available which has an `expandRanges()` function, would that be useful here?

Reply:
expanding the ranges can cause performance issue when range is too long eg. 1...100000, this was an issue in beam before and the solution was just to use between `1` and `1000000` instead of in()

Reply:
Sounds like this approach could also be added to the trait for potentially using in Beam or other places too.

Reply:
There's actually one trait on beam that uses this Enjin\Platform\Beam\Rules\Traits\IntegerRange. We can make the regex static, but will need to move this from beam to core. It can be done on separate ticket for refactor.